test/export/StructureDefinitionExporter.test.ts

```diff
@@ -1989,6 +1989,46 @@ describe('StructureDefinitionExporter', () => {
     );
   });
 
+  it('should resolve soft indexing within Caret Paths on profiles', () => {
+    // Profile: TestPatient
+    // Parent: Patient
+    // * address.line ^slicing.discriminator[+].type = #pattern
+    // * address.line ^slicing.discriminator[=].path = "$this"
+    // * address.line ^slicing.rules = #open
+    // * address.line contains ApartmentName 0..1
+
+    const testPatient = new Profile('TestPatient');
+    testPatient.parent = 'Patient';
+
+    const slicingTypeLine = new CaretValueRule('address.line');
+    slicingTypeLine.caretPath = 'slicing.discriminator[+].type';
+    slicingTypeLine.value = new FshCode('pattern');
+    const slicingPathLine = new CaretValueRule('address.line');
+    slicingPathLine.caretPath = 'slicing.discriminator[=].path';
+    slicingPathLine.value = 'code';
+    const slicingRulesLine = new CaretValueRule('address.line');
+    slicingRulesLine.caretPath = 'slicing.rules';
+    slicingRulesLine.value = new FshCode('open');
+    const containsAddressLine = new ContainsRule('address.line');
+    containsAddressLine.items.push({ name: 'ApartmentName' });
+    const apartmentNameCard = new CardRule('address.line[ApartmentName]');
+    apartmentNameCard.min = 0;
+    apartmentNameCard.max = '1';
+
+    testPatient.rules.push(
+      slicingTypeLine,
+      slicingPathLine,
+      slicingRulesLine,
+      containsAddressLine,
+      apartmentNameCard
+    );
+
+    exporter.exportStructDef(testPatient);
+    const sd = pkg.profiles[0];
+    const apartmentNameElement = sd.findElement('Patient.address.line:ApartmentName');
+    expect(apartmentNameElement).toBeDefined();
```

Comment:
I think it would be more clear that you're testing the soft indexing if you checked the values of `apartmentNameElement.slicing.discriminator[0].type` and `apartmentNameElement.slicing.discriminator[0].path`.

Reply:
Addressed in the most recent commit! The `slicing` property was held in `Patent.address.line` element, rather than in the sliced element itself.